dotcom-rendering/src/types/renderingTarget.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * An enum defining what where we are targeting a particular rendered
+ * page to be shown
+ *
+ * This can be used to make decisions during rendering, where there
+ * might be differences in the requirements of each target.
+ *
+ * Targets:
+ * - Web 	=> A full web browser, such as chrome or safari on a desktop computer, laptop or phone.
+ * - Apps 	=> A webview rendered within the Android or iOS live apps
+ */
+export enum RenderingTarget {
+	Web,
+	Apps,
+}
```

Comment:
As [I mentioned in the RFC](https://github.com/guardian/dotcom-rendering/issues/7256#issuecomment-1462488634), I would recommend avoiding TypeScript enums, as they “are one of the few features TypeScript has which is not a type-level extension of JavaScript”[^1].

To prove my point, what do you expect the value of `RenderingTarget[0]` to be? What about `RenderingTarget.Web === 0`?

[^1]: https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
Would your suggestion be to use a string union? What's the best alternative

Reply:
Yes, string unions are a lot less confusing. If you need to do custom type-guard, the pattern I shared in the RFC works well:

> ```ts
> // If you define a array `as const`…
> const renderingTarget = [
>     "Web",
>     "Apps",
>     "Editions",
> ] as const;
> 
> // … you can infer the type from it…
> export type RenderingTarget = renderingTarget[number];
> 
> // … and create a custom type guard, if you so desire!
> export isRenderingTarget = (target: string): target is RenderingTarget =>
> 	renderingTarget.map(String).includes(target)
> ```